dbcontrollers/resources.go

```diff
@@ -0,0 +1,190 @@
+package dbcontrollers
+
+import (
+	"database/sql"
+	"errors"
+	"strings"
+
+	"github.com/artofimagination/mysql-resources-db-go-service/models"
+	"github.com/artofimagination/mysql-resources-db-go-service/mysqldb"
+	"github.com/google/uuid"
+)
+
+var ErrResourceNotFound = errors.New("The selected resource not found")
+var ErrResourceAlreadyExists = errors.New("The resource already exists")
+var ErrResourceHasTooManyAttachements = errors.New("The resource has too many attachements")
+
+var ErrDuplicateEntrySubString = "Duplicate entry"
+
+// MaxContentItems describes the maximum number or resources to upload to a resources an attachement
+var MaxContentItems = 2
+
+func (c *MYSQLController) AddResource(resource *models.Resource) error {
+	tx, err := c.DBConnector.ConnectSystem()
```

Comment:
Don't you need to close the transaction when leave this function?
e.g.: `defer tx.Close()`

Reply:
there are tx.Commit() or Rollbacks in place to finish the transaction. In theory the functional tests make sure that commits/rollbacks are in place, otherwise the changes would not happen and the tests would not find the changed data.